frame/dapps-staking/src/pallet/mod.rs

```diff
@@ -820,5 +837,11 @@ pub mod pallet {
                 }
             }
         }
+
+        /// Distribute era's rewards to all registered contracts
+        /// The block rewards are accumulated on the pallets's account during an era
+        ///
+        /// This is called at the end of each Era
+        fn distribute_era_rewards() {}
```

Comment:
I suggest to take balance snapshot instead of reward distribution, and then estimate reward in each staker call separately.

Reply:
renamed. in the next commit